SavedInstances/Core/Core.lua

```diff
@@ -2537,6 +2570,15 @@ function SI:toonInit()
   -- real update comes later in UpdateToonData
   ti.DailyResetTime = ti.DailyResetTime or SI:GetNextDailyResetTime()
   ti.WeeklyResetTime = ti.WeeklyResetTime or SI:GetNextWeeklyResetTime()
+
+  local _, specName1 = GetSpecializationInfo(1)
+  ti.Spec1Name = specName1 or ti.Spec1Name
+  local _, specName2 = GetSpecializationInfo(2)
+  ti.Spec2Name = specName2 or ti.Spec2Name
+  local _, specName3 = GetSpecializationInfo(3)
+  ti.Spec3Name = specName3 or ti.Spec3Name
+  local _, specName4 = GetSpecializationInfo(4)
+  ti.Spec4Name = specName4 or ti.Spec4Name
```

Comment:
I'd prefer storing SpecializationID and get the name when required. You can use `GetSpecializationInfoForSpecID`.

Reply:
I agree. I've updated this!

I also moved this code, as apparently I cannot call `GetSpecializationInfo(1)` inside the `SI:toonInit()` function for some reason. I didn't realize this because those parameters were cached locally